python/google/protobuf/internal/well_known_types.py

```diff
@@ -244,6 +244,34 @@ def FromDatetime(self, dt):
     self.seconds = calendar.timegm(dt.utctimetuple())
     self.nanos = dt.microsecond * _NANOS_PER_MICROSECOND
 
+  def _comparable_scalar(self, obj):
+    if isinstance(obj, datetime):
+      return calendar.timegm(obj.utctimetuple()) * _NANOS_PER_SECOND + obj.microsecond * _NANOS_PER_MICROSECOND
+    return obj.ToNanoseconds()
+
+  def timetuple(self):
+    """Only exists to allow comparisons to datetimes in Python 2.7"""
```

Comment:
[[docs](https://docs.python.org/2/library/datetime.html#datetime.datetime.tzinfo)]
> **Note:** In order to stop comparison from falling back to the default scheme of comparing object addresses, datetime comparison normally raises TypeError if the other comparand isn’t also a datetime object. However, NotImplemented is returned instead if the other comparand has a timetuple() attribute. This hook gives other kinds of date objects a chance at implementing mixed-type comparison. If not, when a datetime object is compared to an object of a different type, TypeError is raised unless the comparison is == or !=. The latter cases return False or True, respectively.

Reply:
Humm, it looks like this will only go down to seconds https://docs.python.org/3/library/time.html#time.struct_time - it may be better to drop this and say it isn't available in Python2